pkg/skaffold/render/renderer/renderer.go

```diff
@@ -167,22 +139,78 @@ func (r *SkaffoldRenderer) Render(ctx context.Context, out io.Writer, builds []g
 				},
 			})
 	}
+	if err := os.RemoveAll(r.hydrationDir); err != nil {
+		return sErrors.NewError(err,
+			proto.ActionableErr{
+				Message: fmt.Sprintf("unable to delete Kptfile in %v", r.hydrationDir),
+				ErrCode: proto.StatusCode_RENDER_KPTFILE_INIT_ERR,
+				Suggestions: []*proto.Suggestion{
+					{
+						SuggestionCode: proto.SuggestionCode_KPTFILE_MANUAL_INIT,
+						Action:         fmt.Sprintf("suggest manually delete `rm -rf %v`", r.hydrationDir),
+					},
+				},
+			})
+	}
```

Comment:
nit: this would be cleaner with this error defined in a separate function, maybe even a separate file:

```golang
func KptfileDeleteError(hydrationDir string)
```

Reply:
Agree, we will move all these errors to the errors/errors.go package. But since this PR is to v2 branch, which does not have the commits for the errors.go package. We would need to cherry-pick the missing PR first and then migrate the error message there. I added some todos in the code.

Reply:
see other comment, you can move them to `render/errors.go`.

Reply:
Done.